### PR TITLE
[INLONG-8751][Manager] Fix the response that was empty for data preview

### LINKED ISSUE #8751: [Bug][Manager] Response of data preview was empty
### What happened

image
The inlong group has been reporting data, but the data preview cannot be queried

### What you expected to happen

return multiple messages normally

### How to reproduce

1、Create an inlong group, select pulsar as MQ, and set the pulsar queue model as serial. 
2、Use sdk to report data,
3、Use the data preview function

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by castorqin**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8751  

### Motivation

fix Data preview query data return is empty

### Modifications
The details can be found at [#8751  ](https://github.com/apache/inlong/issues/8751).

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 1]
```diff
@@ -389,27 +390,50 @@ private boolean subscriptionExists(PulsarAdmin pulsarAdmin, String topic, String
...
+        try {
+            PartitionedTopicMetadata partitionedTopicMetadata = pulsarAdmin.topics()
+                    .getPartitionedTopicMetadata(topicFullName);
+            int partitionCount = partitionedTopicMetadata.partitions;
+            boolean serial = partitionCount == 0;
+            for (int messageIndex = 0; messageIndex < messageCount; messageIndex++) {
+                int currentPartitionNum = messageIndex % partitionCount;
+                int messagePosition = messageIndex / partitionCount;
+                String topicNameOfPartition = buildTopicNameOfPartition(topicFullName, currentPartitionNum, serial);
+                messageList.addAll(queryMessageFromPulsar(topicNameOfPartition, pulsarAdmin, messageIndex,
```
[the remark this answers is not in the document]
Comment (castorqin) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java]: done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 2]
```diff
@@ -389,27 +390,65 @@ private boolean subscriptionExists(PulsarAdmin pulsarAdmin, String topic, String
...
+            LOGGER.error(errMsg, e);
+            throw new BusinessException(errMsg + e.getMessage());
+        }
+        return partitionedTopicMetadata.partitions;
+    }
+
+    /**
+     * Use pulsar admin to query message
+     */
+    private List<BriefMQMessage> queryMessageFromPulsar(String topic, PulsarAdmin pulsarAdmin, int index,
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java:427]: topic -> topicPartition

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 3]
```diff
@@ -389,27 +390,65 @@ private boolean subscriptionExists(PulsarAdmin pulsarAdmin, String topic, String
...
+            Message<byte[]> pulsarMessage = pulsarAdmin.topics().examineMessage(topic, "latest", messagePosition);
+            Map<String, String> headers = pulsarMessage.getProperties();
+            int wrapTypeId = Integer.parseInt(headers.getOrDefault(InlongConstants.MSG_ENCODE_VER,
+                    Integer.toString(DataProxyMsgEncType.MSG_ENCODE_TYPE_INLONGMSG.getId())));
+            DeserializeOperator deserializeOperator = deserializeOperatorFactory.getInstance(
+                    DataProxyMsgEncType.valueOf(wrapTypeId));
+            briefMQMessages.addAll(deserializeOperator.decodeMsg(streamInfo, pulsarMessage.getData(),
+                    headers, index));
+        } catch (Exception e) {
+            LOGGER.error("query message from pulsar error", e);
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java:440]: warn

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java [hunk 4]
```diff
@@ -387,29 +388,67 @@ private boolean subscriptionExists(PulsarAdmin pulsarAdmin, String topic, String
...
+                    pulsarAdmin.topics().examineMessage(topicPartition, "latest", messagePosition);
+            Map<String, String> headers = pulsarMessage.getProperties();
+            int wrapTypeId = Integer.parseInt(headers.getOrDefault(InlongConstants.MSG_ENCODE_VER,
+                    Integer.toString(DataProxyMsgEncType.MSG_ENCODE_TYPE_INLONGMSG.getId())));
+            DeserializeOperator deserializeOperator = deserializeOperatorFactory.getInstance(
+                    DataProxyMsgEncType.valueOf(wrapTypeId));
+            briefMQMessages.addAll(deserializeOperator.decodeMsg(streamInfo, pulsarMessage.getData(),
+                    headers, index));
+        } catch (Exception e) {
+            LOGGER.warn("group {} stream {} topic{},query message from pulsar error", streamInfo.getInlongGroupId(),
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarOperator.java:439]: query message from pulsar error for groupId = {}, streamId = {}

**Review by fuweng11 — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**